examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
+    except AssertionError as e:
+        print(f'invalid file extension .{split_filename[1]}')
+        raise e
+    if split_filename[1] != 'json':
+        print(f'parse excel input to {json_filename}')
+        convert_file(input_filename)
+
+    json_data = load_json(json_filename)
+    read_eqpt_library(EQPT_LIBRARY_FILENAME)
```

Comment:
We probably don't need `eqpt_library` to be a global variable here.

Prefer to keep state management very direct:
`equipment = read_equipment(filename)`

Rather than hardcoding the `EQPT_LIBRARY_FILENAME` path, this can be an argument provided to the argument parser with a default.

Reply:
no, we don't want to read a possibly very large json file everytime we look up for an eqpt.

Reply:
We should be able to load the equipment information just once while still avoiding the use of global state.

Reply:
Please help, 
the global variable is the best I could find for this purpose.